File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.c
int lengthOfLongestSubstring(char* s) {
    int i = 0, j = 0;
    int hashSet[256] = {0};
    int maxLength = 0;
    while (s[j] != '\0') {
        if (hashSet[s[j]] == 0) {
            hashSet[s[j]] = 1;
            maxLength = (j - i + 1 > maxLength) ? (j - i + 1) : maxLength;
            j++;
        } else {
            hashSet[s[i]] = 0;
            i++;
        }
    }
    return maxLength;
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s):
        max_len = 0

        for i in range(len(s)):
            temp = ""

            for j in range(i, len(s)):
                if s[j] not in temp:
                    temp += s[j]
                    max_len = max(max_len, len(temp))
                else:
                    break

        return max_len
        